pkg/skaffold/debug/debug.go

```diff
@@ -17,145 +17,21 @@ limitations under the License.
-	"k8s.io/apimachinery/pkg/runtime/serializer/json"
-	"k8s.io/client-go/kubernetes/scheme"
-
-	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
-	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/graph"
-	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/manifest"
-	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/runcontext"
-)
-
-var (
-	decodeFromYaml = scheme.Codecs.UniversalDeserializer().Decode
-	encodeAsYaml   = func(o runtime.Object) ([]byte, error) {
-		s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
-		var b bytes.Buffer
-		w := bufio.NewWriter(&b)
-		if err := s.Encode(o, w); err != nil {
-			return nil, err
-		}
-		w.Flush()
-		return b.Bytes(), nil
-	}
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/config"
 )
 
-// ApplyDebuggingTransforms applies language-platform-specific transforms to a list of manifests.
-func ApplyDebuggingTransforms(l manifest.ManifestList, builds []graph.Artifact, registries manifest.Registries) (manifest.ManifestList, error) {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
-	retriever := func(image string) (imageConfiguration, error) {
-		if artifact := findArtifact(image, builds); artifact != nil {
-			return retrieveImageConfiguration(ctx, artifact, registries.InsecureRegistries)
-		}
-		return imageConfiguration{}, fmt.Errorf("no build artifact for %q", image)
-	}
-	return applyDebuggingTransforms(l, retriever, registries.DebugHelpersRegistry)
+type Config interface {
+	Mode() config.RunMode
 }
 
-func applyDebuggingTransforms(l manifest.ManifestList, retriever configurationRetriever, debugHelpersRegistry string) (manifest.ManifestList, error) {
-	var updated manifest.ManifestList
-	for _, manifest := range l {
-		obj, _, err := decodeFromYaml(manifest, nil, nil)
-		if err != nil {
-			logrus.Debugf("Unable to interpret manifest for debugging: %v\n", err)
-		} else if transformManifest(obj, retriever, debugHelpersRegistry) {
-			manifest, err = encodeAsYaml(obj)
-			if err != nil {
-				return nil, fmt.Errorf("marshalling yaml: %w", err)
-			}
-			if logrus.IsLevelEnabled(logrus.DebugLevel) {
-				logrus.Debugln("Applied debugging transform:\n", string(manifest))
-			}
-		}
-		updated = append(updated, manifest)
-	}
-
-	return updated, nil
-}
-
-// findArtifact finds the corresponding artifact for the given image.
-// If `builds` is empty, then treat all `image` images as a build artifact.
-func findArtifact(image string, builds []graph.Artifact) *graph.Artifact {
-	if len(builds) == 0 {
-		logrus.Debugf("No build artifacts specified: using image as-is %q", image)
-		return &graph.Artifact{ImageName: image, Tag: image}
-	}
-	for _, artifact := range builds {
-		if image == artifact.ImageName || image == artifact.Tag {
-			logrus.Debugf("Found artifact for image %q", image)
-			return &artifact
-		}
-	}
-	return nil
+type Debugger interface {
+	Start(context.Context, []string) error
```

Comment:
can you document what `Debugger` is and what `Start` and `Stop` are supposed to do?